### PR TITLE
log_array added to logger

### PR DESCRIPTION
**Pull request by jdahlblom**
* Default values for log_table added
* log_table public as it is useful every now and then
* log_array logs index and value only, name implies how it logs
* log_indexes and log_array are both for debugging

### CODE REVIEW

File: Scripts/DCS-BIOS/lib/common/Logger.lua [hunk 1]
```diff
@@ -146,7 +146,6 @@ end
 -- Break ALL when max depth reached
 -- Break when data logged exceeds limit
 
---- @private
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Logger.lua:149]: Why was this removed?

File: Scripts/DCS-BIOS/lib/common/Logger.lua [hunk 2]
```diff
@@ -236,4 +235,43 @@ function Logger:log_table_indexes(tab)
 	end
 end
 
+--- Logs an array
+--- @require array array to log
+function Logger:log_array(array)
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Logger.lua:240]: Shouldn't this have a log level that is passed through from like log_info to log_error?

File: Scripts/DCS-BIOS/lib/common/Logger.lua [hunk 3]
```diff
@@ -53,7 +53,7 @@ function Logger:log(level, obj)
 
 	if type(obj) == "table" then
 		self:log_simple(level, "Logging table, consider using setting max_bytes_to_log and call log_table() directly.")
-		self:log_table(level, obj, 10)
+		self:log_table(level, obj, 40)
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Logger.lua:56]: Is there any way we can distinguish an array from a table? Guessing not in lua...

Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/common/Logger.lua:56]: Well this is the problem I was having, I tried to log the parse_indication with the original function and nothing came out. I guess I have to do some local testing.

File: Scripts/DCS-BIOS/lib/common/Logger.lua (hunk 2 above)
Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/common/Logger.lua:240]: This was thought to be a debug function

File: Scripts/DCS-BIOS/lib/common/Logger.lua (hunk 1 above)
Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/common/Logger.lua:149]: I tried calling it directly when I wasn't getting any output. Will change.

Comment (charliefoxtwo) [reply Scripts/DCS-BIOS/lib/common/Logger.lua:149]: Ah the price annotation just affects intellisense for the lua language server plugin (and it will yell at you if you try to use it outside of the defined scope) but it doesn't affect runtime behavior

File: Scripts/DCS-BIOS/lib/common/Logger.lua (hunk 3 above)
Comment (charliefoxtwo) [reply Scripts/DCS-BIOS/lib/common/Logger.lua:59]: The only result of parse_indication which has an array is the KA-50 iirc (I may have removed that too, I can't recall atm)
